posturemanagementv1/posture_management_v1_integration_test.go

```diff
@@ -0,0 +1,226 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package posturemanagementv1_test
+
+import (
+	"fmt"
+	"github.com/IBM/go-sdk-core/v5/core"
+	examples "github.com/ibm-cloud-security/scc-go-sdk/examples/posturemanagementv1"
+	scc "github.com/ibm-cloud-security/scc-go-sdk/posturemanagementv1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"os"
+)
+
+/**
+ * This file contains an integration test for the Posture Management v1 package.
+ *
+ * Notes:
+ *
+ * The integration test will automatically skip tests if the required config file is not available.
+ */
+
+var (
+	err           error
+	config        map[string]string
+	collectorIds  []string
+	credentialId  string
+	accountId     string
+	apiKey        string
+	authUrl       string
+	apiUrl        string
+	scopeId       *string
+	scanId        string
+	scopeName     *string
+	authenticator core.IamAuthenticator
+	options       scc.PostureManagementV1Options
+)
+
+var _ = Describe(`SCC test`, func() {
+	const externalConfigFile = "../posture_management_v1.env"
+
+	Describe(`Demo`, func() {
+
+		BeforeEach(func() {
+			_, err = os.Stat(externalConfigFile)
+			if err != nil {
+				Skip("External configuration file not found, skipping tests: " + err.Error())
+			}
+
+			err = os.Setenv("IBM_CREDENTIALS_FILE", externalConfigFile)
+			if err != nil {
+				return
+			}
+			config, err = core.GetServiceProperties(scc.DefaultServiceName)
+			if err != nil {
+				Skip("Error loading service properties, skipping tests: " + err.Error())
+			}
+
+			accountId = config["ACCOUNT_ID"]
+			if accountId == "" {
+				Skip("Unable to load account configuration property, skipping tests")
+			}
+
+			apiKey = config["IAM_API_KEY"]
+			if apiKey == "" {
+				Skip("Unable to load API key configuration property, skipping tests")
+			}
+
+			authUrl = config["IAM_APIKEY_URL"]
+			if authUrl == "" {
+				Skip("Unable to load auth URL configuration property, skipping tests")
+			}
+
+			apiUrl = config["API_URL"]
+			if apiUrl == "" {
+				Skip("Unable to load API URL configuration property, skipping tests")
+			}
+
+			authenticator = core.IamAuthenticator{
+				ApiKey: apiKey,
+				URL:    authUrl,
+			}
+
+			options = scc.PostureManagementV1Options{
+				Authenticator: &authenticator,
+				URL:           apiUrl,
+			}
+
+		})
+
+		//TODO fix collector issue -- skip for now
+		It(`Create Collector`, func() {
+			fmt.Println(`Create Collector`)
+			statusCode, id := examples.CreateCollector(options, accountId)
+			Expect(statusCode).To(Equal(201))
+			Expect(id).NotTo(BeNil())
+		})
+
+		//new credential to be created
+		It(`Create Credential`, func() {
+			fmt.Println(`Create Credential`)
+			var statusCode int
+			credentialPath := config["CREDENTIAL_PATH"]
+			pemPath := config["PEM_PATH"]
```

Comment:
I assume the PEM_PATH and CREDENTIAL_PATH are exposed as 

POSTURE_MANAGEMENT_PEM_PATH and  POSTURE_MANAGEMENT_CREDENTIAL_PATH in the value of POSTURE_MANAGEMENT_ENV set in travis. Then we don't need

```
	echo "${POSTURE_PEM_ENV}" | base64 -d >> posture-credential.pem
       echo "${POSTURE_CREDENTIAL_ENV}" | base64 -d >> posture-credential.json
```
in build/testScript.sh

Reply:
@ashishth09 the CREDENTIAL and PEM are the file contents. The create credentials API needs to read from those files and sends them as multipart/form-data. That is why we need the path and contents.

Reply:
Where are these files read in the tests cases posture-credential.pem and posture-credential.json? I couldn't find them.

Reply:
> Where are these files read in the tests cases posture-credential.pem and posture-credential.json? I couldn't find them.

those files are generated in `testScript.sh`

Reply:
you cannot use external files or examples in ITs.... just fix test cases that are there in generated ITs and if required you can add more test cases.

Reply:
accepted and will update accordingly